lib/config/presets/index.ts

```diff
@@ -127,6 +128,16 @@ export function parsePreset(input: string): ParsedPreset {
     } else {
       presetName = str.slice(1);
     }
+  } else if (str.includes('//')) {
+    // non-scoped namespace with a subdirectory preset
+    [, packageName, presetPath, presetName] = /(.*?)\/\/(?:(.*)\/)?(.*)$/.exec(
+      str
+    );
+
+    // Validation
+    if (presetPath?.includes(':') || presetName.includes(':')) {
+      throw new Error('prohibited sub-preset');
+    }
```

Comment:
```suggestion
    // non-scoped namespace with a subdirectory preset
    const re = /^([\w/]+?)\/\/(?:([\w/]+)\/)?(\w+)$/; // todo: perf, move to module scope
    // Validation
    if (!re.test(str) || str.includes(':')) throw Error('invalid preset'); // todo: format
    [, packageName, presetPath, presetName] = re.exec(
      str
    );
```

I thought a more strict regex to only allow url /path like strings.

https://regex101.com/r/VPyjFN/1

Reply:
That regex wouldn't work for any repository that has a period in its name or path to the preset file. Since the info pulled out by the regex is just passed back to an API for the provider, I think it would be okay to trust that the user is inputting a valid string to point to whatever preset they want to load (so long as it contains `//`). That seems like a better option compared to potentially marking valid configs as invalid.

Reply:
I'm strictly against too open regex, as they can be exploited to an dos. I'll update my suggestion tomorrow.

Reply:
```js
/^([\w/]+?)\/\/(?:([\w\./]+)\/)?(\w.+)$/
```

https://regex101.com/r/VzHJNO/1

Reply:
Just pushed a fix to use the above regex with a single modification to allow periods in the repo name